### ISSUE TITLE
KEY_CHANGE_DICT for retrocompatibility will fail for non-unique key names

### ISSUE THREAD

**Issue by ahill187** [ivadomed/ivadomed#670]
This issue was mentioned here:  https://github.com/ivadomed/ivadomed/pull/668#pullrequestreview-584541738

### Current behavior

In `config_manager.py`, there is a `KEY_CHANGE_DICT`, which allows for retrocompatibility if/when we change the key names in the config file. For example, if I use a config file that uses the old key `UNet3D`, the `KEY_CHANGE_DICT` maps this to the new key and updates the config file accordingly:

```KEY_CHANGE_DICT = {'UNet3D': 'Modified3DUNet'}```

However, because the `KEY_CHANGE_DICT` does not specify the level of the JSON dict, and because nested dictionary keys do not need to be unique, the edge case where we needed to change a non-unique key would fail.

For example, say this was in our original config file:
```
    "training_parameters": {
        "loss": {
            "name": "DiceLoss"
        },
        "scheduler": {
            "initial_lr": 0.001,
            "lr_scheduler": {
                "name": "CosineAnnealingLR",
                "base_lr": 1e-5,
                "max_lr": 1e-2
            }
        }
```

Suppose we want to rename the key for `name` in the `loss` dictionary to `loss_name`. We could update the `KEY_CHANGE_DICT` as follows:

```KEY_CHANGE_DICT = {'UNet3D': 'Modified3DUNet', 'name': 'loss_name'}```

But then we run into the issue: which `name` key are we referring to? Do we mean the `name` key associated with the `loss` dictionary, or do we mean the `name` key associated with the `lr_scheduler` dictionary? In this example, we may end up updating the incorrect key.

## Issue Solution

I think we somehow need to add some information into the `KEY_CHANGE_DICT` to specify which part of the JSON tree we intend to update.

**Comment by mariehbourget** [comment]
As discussed in #699, I added another [list](https://github.com/ivadomed/ivadomed/blob/a56e9d97ef935af0cdfab17d66e3a3e7b8d5debe/ivadomed/config_manager.py#L70) `KEY_SPLIT_DATASET_CHANGE_LST` and [function](https://github.com/ivadomed/ivadomed/blob/a56e9d97ef935af0cdfab17d66e3a3e7b8d5debe/ivadomed/config_manager.py#L132) `change_keys_values` to handle key/value changes for old split dataset params.

There is probably a better to do this. As a reference, the key/value change is:

- `"method": "per_patient"` --> `"split_method" : "participant_id"`
- `"method": "per_center"` --> `"split_method" : "participant_id"` and `"data_testing": {"data_type": "institution_id", "data_value":[]}`
- `"method": "per_center"` and `"center_test": [LIST_CONTENT]` --> `"split_method" : "participant_id"` and `"data_testing": {"data_type": "institution_id", "data_value":[LIST_CONTENT]}`
